examples/boids_spatial3D/src/main.cu

```diff
@@ -266,6 +264,36 @@ FLAMEGPU_AGENT_FUNCTION(inputdata, flamegpu::MessageSpatial3D, flamegpu::Message
         vec3Div(agent_fx, agent_fy, agent_fz, agent_fscale);
     }
 
+    float minSpeed = 0.5f;
+    if (agent_fscale < minSpeed) {
+        // Normalise
+        vec3Div(agent_fx, agent_fy, agent_fz, agent_fscale);
+        // Scale to min
+        vec3Mult(agent_fx, agent_fy, agent_fz, minSpeed);
+    }
+   
+    // Steer away from walls - Computed post normalization to ensure good avoidance. Prevents constant term getting swamped
+    float wallInteractionDistance = 0.10f;
+    float wallSteerStrength = 0.05f;
+    if (agent_x - FLAMEGPU->environment.getProperty<float>("MIN_POSITION") < wallInteractionDistance) {
```

Comment:
I'd suggest storing `MIN_POSITION` and `MAX_POSITION` in `const float`, both to improve readability and just incase the compiler doesn't realise it can safely remove multiple accesses through curve (It should but who knows what it will do to register usage in 11.3+)